### PR TITLE
Remove duplicate views: Show notices

### LINKED ISSUE #97400: Remove duplicate views: Show notices
See pekYwv-4QW-p2

We need to show a notice when the WP Admin view is enforced for users participating in the remove duplicate views experiment.

Image

We need to show them only to users currently using the default admin interface, and in every screen that is enforced. Each notice can be permanently dismissed.

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/wp-calypso/issues/97400

## Proposed changes:

Displays a notice when a user visits the enforced WP Admin view of a removed Calypso screen for the first time.

Screenshot 2024-12-13 at 16 47 57

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Go to the placeholder experiment page: 22124-explat-experiment
- Assign yourself to the `treatment` variation using the bookmarklets
- Visit any of the screens that are currently enforcing the WP Admin view:
  - Posts > All Posts
  - Posts > Tags
  - Posts > Categories
  - Portfolio > All Projects
  - Testimonials > All Testimonials
- Make sure a notice shows up correctly
- Make sure the notice does not show up in all the following scenarios:
  - Already dismissed
  - When visiting the WP Admin view if your admin interface was already set to `wp-admin` (i.e. untangled)
  - When visiting the WP Admin view if your preferred view is `classic` (i.e. tangled classic)
  - When visiting the WP Admin view while being assigned to the `control` variation

### CODE REVIEW

**Comment by rcrdortiz** [comment]
The notice gets closed once we click outside of it. Since once dismissed it's not shown again, should we disable clicking outside the main CTA to dismiss the notice?

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-interface/wpcom-admin-interface.php [hunk 1]
```diff
@@ -416,3 +416,77 @@ function wpcom_is_duplicate_views_experiment_enabled() {
...
+	add_filter( 'pre_option_wpcom_admin_interface', 'wpcom_admin_interface_pre_get_option', 10 );
+	if ( $uses_wp_admin_interface ) {
+		return;
+	}
+
+	remove_filter( 'get_user_option_jetpack_admin_menu_preferred_views', 'wpcom_admin_get_user_option_jetpack' );
+	$preferred_views = get_user_option( 'jetpack_admin_menu_preferred_views' );
+	add_filter( 'get_user_option_jetpack_admin_menu_preferred_views', 'wpcom_admin_get_user_option_jetpack' );
+	if ( ! empty( $preferred_views ) && isset( $preferred_views[ $current_screen ] ) && $preferred_views[ $current_screen ] === 'classic' ) {
+		return;
```
Comment (rcrdortiz) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-interface/wpcom-admin-interface.php:471]: Just out of curiosity, what happens if we don't remove the filers before getting the options?

**Review by rcrdortiz — DISMISSED** [review]
Everything works as expected.

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-interface/wpcom-admin-interface.php (hunk 1 above)
Comment (mmtr) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-interface/wpcom-admin-interface.php:471]: The notices would be displayed to users currently using the classic view (i.e. tangled wp-admin) or the classic admin interface (i.e. untangled wp-admin).

**Comment by mmtr** [comment]
> The notice gets closed once we click outside of it. Since once dismissed it's not shown again, should we disable clicking outside the main CTA to dismiss the notice?

I noticed that too, yeah, not ideal. But the `Guide` component does not support having a different callback for when we click outside, so I don't think we have any alternative here.

**Pull request merged**